### PR TITLE
Feat:line items ts

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- convert `LineItemService` to typescript

### CODE REVIEW

File: packages/medusa/src/services/line-item.ts [hunk 1]
```diff
@@ -138,14 +176,13 @@ class LineItemService extends BaseService {
 
   /**
    * Create a line item
-   * @param {LineItem} lineItem - the line item object to create
-   * @return {LineItem} the created line item
+   * @param {DeepPartial<LineItem>} lineItem - the line item object to create
+   * @return {Promise<LineItem>} the created line item
    */
-  async create(lineItem) {
+  async create(lineItem: DeepPartial<LineItem>): Promise<LineItem> {
```
Comment (olivermrbl) [inline packages/medusa/src/services/line-item.ts:182]: nit: To be consistent with what we've done in already converted services, we could define a `CreateLineItemProps`.

File: packages/medusa/src/services/line-item.ts [hunk 2]
```diff
@@ -159,11 +196,10 @@ class LineItemService extends BaseService {
    * @param {object} update - the properties to update on line item
    * @return {LineItem} the update line item
    */
-  async update(id, update) {
+  async update(id: string, update: Partial<LineItem>): Promise<LineItem> {
```
Comment (olivermrbl) [inline packages/medusa/src/services/line-item.ts:199]: nit: Same as above. Could define `UpdateLineItemProps`

File: packages/medusa/src/services/line-item.ts [hunk 3]
```diff
@@ -174,24 +210,55 @@ class LineItemService extends BaseService {
...
+
+      lineItems.forEach((li) => {
+        li.has_shipping =
+          shippingProfiles.find(
+            (sp) => sp === li.variant?.product?.profile_id
+          ) !== undefined
+      })
+
+      const result = await lineItemRepository.save(lineItems)
+      return result
```
Comment (olivermrbl) [inline packages/medusa/src/services/line-item.ts:249]: See comment from @srindom [here](https://github.com/medusajs/medusa/pull/912#discussion_r769013069)

File: packages/medusa/src/types/line-item.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+export type GenerateConfig = {
+  unit_price?: number
+  metadata?: object
+}
+
+export type createLineItemDto = {
```
Comment (olivermrbl) [inline packages/medusa/src/types/line-item.ts:6]: very nit: Should be capitalized and renamed to be consistent with other services: `CreateLineItemInput`

File: packages/medusa/src/types/line-item.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+  cart_id?: string
+  order_id?: string
+  swap_id?: string
+  claim_order_id?: string
+
+  metadata?: object
+}
+
+export type updateLineItemDto = {
```
Comment (olivermrbl) [inline packages/medusa/src/types/line-item.ts:29]: very nit: Should be capitalized and renamed to be consistent with other services: `UpdateLineItemInput`

**Review by olivermrbl — APPROVED** [review]
LGTM w. small comments 😄 

general nit: Remove types from JSDoc

**Comment by srindom** [comment]
line-item is migrated so closing this :)

**Pull request closed without merging**